rxfingerprint/src/main/java/com/mtramin/rxfingerprint/FingerprintEncryptionObservable.java

```diff
@@ -115,7 +128,7 @@ protected void onAuthenticationSucceeded(Subscriber<FingerprintEncryptionResult>
 			subscriber.onNext(new FingerprintEncryptionResult(FingerprintResult.AUTHENTICATED, null, encryptedString));
 			subscriber.onCompleted();
 		} catch (CryptoDataException | IllegalBlockSizeException | BadPaddingException | InvalidParameterSpecException | UnsupportedEncodingException e) {
-			subscriber.onError(e);
+			checkKey(e, subscriber, () -> subscriber.onNext(new FingerprintEncryptionResult(FingerprintResult.INVALIDATED, null, null)));
```

Comment:
Why does encryption return a different result from this error case than decryption? Here we return `FingerprintResult.INVALIDATED` whereas decryption still throws `KeyPermanentlyInvalidatedException`.